From fdec9090c8de8b2d1edc9f88a77e100d6a4bd8f7 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:11:02 -0400
Subject: [usb] pl2303: fixed handling of CS5 setting

Message-id: <1410887463-20674-591-git-send-email-dzickus@redhat.com>
Patchwork-id: 95547
O-Subject: [RHEL7 PATCH 590/591] USB: pl2303: fixed handling of CS5 setting
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit a313249937820f8b1996133fc285efbd6aad2c5b
Author: Colin Leitner <colin.leitner@googlemail.com>
Date:   Mon Nov 4 19:40:43 2013 +0100

    USB: pl2303: fixed handling of CS5 setting

    This patch fixes the CS5 setting on the PL2303 USB-to-serial devices. CS5 has a
    value of 0 and the CSIZE setting has been skipped altogether by the enclosing
    if. Tested on 3.11.6 and the scope shows the correct output after the fix has
    been applied.

    Tagged to be added to stable, because it fixes a user visible driver bug and is
    simple enough to backport easily.

    Signed-off-by: Colin Leitner <colin.leitner@gmail.com>
    Signed-off-by: Johan Hovold <jhovold@gmail.com>
    Cc: stable <stable@vger.kernel.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/serial/pl2303.c b/drivers/usb/serial/pl2303.c
index 96c7240..b3d5a35 100644
--- a/drivers/usb/serial/pl2303.c
+++ b/drivers/usb/serial/pl2303.c
@@ -480,23 +480,21 @@ static void pl2303_set_termios(struct tty_struct *tty,
 
  pl2303_get_line_request(port, buf);
 
- if (C_CSIZE(tty)) {
-  switch (C_CSIZE(tty)) {
-  case CS5:
-   buf[6] = 5;
-   break;
-  case CS6:
-   buf[6] = 6;
-   break;
-  case CS7:
-   buf[6] = 7;
-   break;
-  default:
-  case CS8:
-   buf[6] = 8;
-  }
-  dev_dbg(&port->dev, "data bits = %d\n", buf[6]);
+ switch (C_CSIZE(tty)) {
+ case CS5:
+  buf[6] = 5;
+  break;
+ case CS6:
+  buf[6] = 6;
+  break;
+ case CS7:
+  buf[6] = 7;
+  break;
+ default:
+ case CS8:
+  buf[6] = 8;
  }
+ dev_dbg(&port->dev, "data bits = %d\n", buf[6]);
 
  /* For reference buf[0]:buf[3] baud rate value */
  pl2303_encode_baud_rate(tty, port, &buf[0]);
-- 
1.7.1